projects/legacy/components/input-password/input-password.component.ts

```diff
@@ -67,7 +67,10 @@ export class TuiInputPasswordComponent
     }
 
     public get inputType(): string {
-        return this.isPasswordHidden || !this.interactive ? 'password' : 'text';
+        return this.isPasswordHidden ||
+            !(this.interactive || (this.readOnly && !this.computedDisabled))
```

Comment:
I believe interactive is a getter for not read-only && not disabled. Not computed disabled is enough to check. Same in template. Also please do this in both v3.x and main.